paper/main.tex

```diff
@@ -1008,7 +1012,35 @@ \subsection{Pitfalls and limitations}
 software would therefore need to use existing non-Davos approaches to managing
 those requirements.
 
-\textcolor{red}{\textbf{TODO: add note about default/fallback project for non-traditional notebook interfaces}}
+While Davos enables developers to conveniently specify all project
+dependencies, there are some edge cases and limitations that are worth
+considering. Many Python packages include (in their setup options) additional
+dependencies that often carry their own version specifications. Although Davos
+will check that the correct version of the requested top-level package is
+installed and imported into the workspace, the version numbers of any
+\textit{dependencies} of the requested package are not checked. In principle,
+this could lead to unexpected behavior, for example if a given package's
+dependencies (or dependencies of those dependencies, etc.) were left
+under-specified. A developer could mitigate this by explicitly smuggling exact
+version numbers of every project dependency (e.g., obtained via pip freeze).
+However, for projects where the versions of dependencies of smuggled packages
+also need to be precisely controlled, a lockfile or a \texttt{requirements.txt}
+file produced by \texttt{pip freeze} (i.e., explicitly specifying \textit{all} packages'
+version numbers) may provide a more comprehensive alternative to Davos.
```

Comment:
I think adding this back in might've been unintentional? We had talked about this, agreed it doesn't make sense, and you had removed it a few months back (and we confirmed that earlier today via Slack as well)

Reply:
thanks for the reminder.  fine to cut this